cloud/cloud.js

```diff
@@ -380,3 +387,105 @@ Moralis.Cloud.define("taskerAbandonTask", async (request) => {
 });
 
 /* ------------------------------------------------------------------- */
+
+Moralis.Cloud.define("createTask", async (request) => {
+  function validateNewTask(newTask, maxReward, maxResponses, config) {
+    const MIN_TASKERS = Number(config.get("NEXT_PUBLIC_MIN_TASKERS"));
+    const MIN_TASK_DATA_CHARS = Number(
+      config.get("NEXT_PUBLIC_MIN_TASK_DATA_CHARS")
+    );
+    const MIN_ETH = Number(config.get("NEXT_PUBLIC_MIN_ETH"));
+
+    if (maxReward < MIN_ETH) return false;
+    if (maxResponses < MIN_TASKERS) return false;
+    if (!Number.isInteger(maxResponses)) return false;
+    if (newTask["title"].length < MIN_TASK_DATA_CHARS) return false;
+    if (newTask["description"].length < MIN_TASK_DATA_CHARS) return false;
+    if (
+      newTask["questions"].some(
+        (question) => question["question"].length < MIN_TASK_DATA_CHARS
+      )
+    )
+      return false;
+    if (
+      newTask["questions"].some((question) => {
+        // See enum QuestionType for type IDs
+
+        // Multiple choice
+        if (question["type"] === 0) {
+          // Check that number of choices is in [1, 5]
+          if (question["options"].length < 1 || question["options"].length > 5)
```

Comment:
should this return false instead because we don't want the number of choices to be < 1 or > 5/

Reply:
Ah, the way `Array.some()` works is that if the passed-in function returns true for at least one element, then `Array.some()` returns `true`. In this case, the passed-in function is designed so that it returns `true` if there is something wrong, making the entire call to `Array.some()` result in `true` and causing `validateNewTask()` to ultimately return `false.